In [2]:
library(tidyverse)


── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [12]:
players <- read_csv("data/players.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Our dataset has 196 rows, meaning we have 196 players with data.

In [10]:
sessions <- read_csv("data/sessions.csv")
sessions

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7    fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8    ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9    96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
11   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
12   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
13   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
14   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
15   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
16   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
17   88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
20   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
21   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
22   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
23   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
25   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
26   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
27   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
28   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
29   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
30   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
⋮    ⋮                                                               
1506 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
1507 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1508 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1509 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
1510 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1511 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1512 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1513 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1514 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1515 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1516 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1517 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
1518 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1519 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
1520 55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1
1521 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1522 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1523 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1524 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1525 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1526 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1527 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1528 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1529 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1530 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1531 36d9cbb4c6